react-forms/my-app/src/RegistrationFormControlled.tsx

```diff
@@ -0,0 +1,35 @@
+import { FormEvent, useState } from 'react';
+
+export function RegistrationFormControlled() {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+
+  function handleSubmit(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    console.log({ username, password });
+  }
+
+  return (
+    <form action="" onSubmit={handleSubmit}>
+      <label htmlFor="">
+        Username:
+        <input
+          type="text"
+          name="username"
```

Comment:
In a controlled form, you don't need to provide the `name` attribute. As above, it doesn't hurt, but leaving it off saves some typing.